core/src/executor/aggregate/mod.rs

```diff
@@ -156,25 +176,47 @@ impl<'a> Aggregator<'a> {
     }
 }
 
-fn aggregate<'a>(
+#[async_recursion(?Send)]
+async fn aggregate<'a>(
     state: State<'a>,
-    context: &BlendContext<'_>,
+    blend_context: Rc<BlendContext<'a>>,
+    filter_context: Option<Rc<FilterContext<'a>>>,
     expr: &'a Expr,
 ) -> Result<State<'a>> {
```

Comment:
```rust
    pub async fn accumulate(
        self,
        filter_context: Option<Rc<FilterContext<'a>>>,
        aggr: &'a Aggregate,
    ) -> Result<State<'a>> {
```

Content to be linked from <https://github.com/gluesql/gluesql/issues/767>
In this function, we keep it as `FilterContext::concat`. Pulling this out to the top